include/EVT/dev/button.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef _EVT_BUTTON_
+#define _EVT_BUTTON_
+
+#include <EVT/io/GPIO.hpp>
+#include <stdint.h>
+
+namespace EVT::core::DEV {
+// Forward declarations:
+// The different pins are hardware specific. Forward declaration to allow
+// at compilation time the decision of which pins should be used.
+
+class Button {
+public:
+    /**
+         * Enum to handle the button trigger states
+         *
+         */
+    enum class LogicLevelOnPress {
+        LOW = 0u,
+        HIGH = 1u,
+    };
+
+    /**
+         * Create an instance of the LED based on the given GPIO pin.
+         *
+         * @param gpio[in] GPIO pin
+         * @param logicLevelOnPress[in] Represents if the button is active high or active low
+         */
+    Button(EVT::core::IO::GPIO& gpio, Button::LogicLevelOnPress logicLevelOnPress);
+
+    /**
+         * Reads the state of the button
+         *
+         * @return The button state.
+         */
+    IO::GPIO::State readButton();
+
+    /**
+         * Time how long the button is in a high state
+         *
+         * @return Time in milliseconds
+         */
+    uint32_t debounce();
```

Comment:
Are the debounce methods implemented? If not I would look into implementing them using [HAL_GetTick](https://stackoverflow.com/questions/42747128/does-hal-gettick-return-ticks-or-milliseconds-and-how-to-measure-in-microsec) method to get the elapsed time since startup in milliseconds.